internal/controllers/machineset/machineset_controller.go

```diff
@@ -313,6 +313,10 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 
 	// Always updates status as machines come up or die.
 	if err := r.updateStatus(ctx, cluster, machineSet, filteredMachines); err != nil {
+		if errors.Is(err, remote.ErrClusterLocked) {
```

Comment:
Let's add a godoc comment that we have to cover this case here because kerrors.NewAggregate doesn't preserve remote.ErrClusterLocked (assuming that is why we do this)